bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/extension/Extension.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.extension;
+
+/**
+ * This class defines an extension.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class Extension {
```

Comment:
My read was that it is one. Is there any other reason to use the type id instead of a type object?

How is this handled in general? Should we rename type in typeId?